content/protocols/pusher.textile

```diff
@@ -0,0 +1,97 @@
+---
+title: Pusher Adapter
+meta_description: "Use the Pusher Adapter to migrate from Pusher to Ably by only changing your API key."
+languages:
+  - javascript
+---
+
+Ably provides the ability to migrate from Pusher to Ably using its Pusher Adapter. The protocol adapter handles all background translation and only requires you to update your API key.
+
+Using an adaptor does add some latency and will be slower than using a native Ably SDK, however the impact is typically in the low milliseconds. It will also be slightly slower than using Pusher natively, but only if you are close to whichever Pusher data center you are using. If you aren't close to the Pusher data center you've chosen, then the extra latency from using the adapter should be more than compensated for by being able to use a data center that is close to you. This is because Ably automatically connects clients to the data center closest to them.
```

Comment:
```suggestion
Using an adapter introduces some latency and is slower than using a native Ably SDK; however, the impact typically remains in the low milliseconds. It will also be slightly slower than using Pusher natively, but only if you are close to whichever Pusher data center you use. If you aren't close to the Pusher data center you've chosen, then the extra latency from using the adapter should be more than compensated for by being able to use a data center that is close to you. This is because Ably automatically connects clients to the data center closest to them.
```

Reply:
Adapter!